### DISCUSSION TITLE
how to auto format afte save

### DISCUSSION THREAD

**Discussion by OriLiMu** [LazyVim/LazyVim#3120] [Q&A] [answered]
here is my config in the config file.
```lua
vim.api.nvim_create_augroup("format json", {})
vim.api.nvim_clear_autocmds({ group = "format json" })
vim.api.nvim_create_autocmd("BufWritePost", {
  pattern = "*.json",
  callback = function()
    vim.lsp.buf.format()
  end,
  group = "format json",
  desc = "format json",
})

```

The json lsp server is installed by mason which is `biome`.

But I still get the error msg

![image](https://github.com/LazyVim/LazyVim/assets/11295355/2e6ad739-90cc-47de-bc36-7c3e2e214fc2)

**Comment by dpetka2001** [comment]
You should configure your formatter of choice in `conform.nvim` and it will auto-format on save.

**Reply by OriLiMu** [reply]
I add command `require("conform").formatters_by_ft = vim.tbl_extend("force", require("conform").formatters_by_ft, { json = { "jq" } })` to the init.lua. But get error conform not found why it happened? 

The conform is built-in I think. I found it in the **Lazy** list

**Reply by dpetka2001** [reply, marked as answer]
You should just create a file `~/.config/nvim/lua/plugins/format.lua` with contents
```lua
return {
  {
    "stevearc/conform.nvim",
    opts = {
      formatters_by_ft = {
        json = { { "jq" } },
      },
    },
  },
}
```
